### ISSUE TITLE
[a11y]: ContainedList [with icons] - screen reader does not identify icons

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13946]
### Package

@carbon/react

### Browser
Safari & FF & iPhone

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology
JAWS & VO

### Description
Icons present in the component

Screen reader does not identify the Icons present in the component

### WCAG 2.1 Violation
n/a

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=&id=components-containedlist--with-icons&viewMode=story

### Steps to reproduce
Record `Number 34` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

**_If_** these icons convey meaningful information, these should be made accessible to the screen reader. Screen reader should identify the image role with appropriate alt text that describes the purpose of each image. 

This needs to be refined by the team

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
There is no guidance on icon usage in lists in the Structured List topic.
I completely agree that where the icon adds meaning to the list, it should have an ALT. I am much less clear on any situation where the icon for a list would be more than decorative.
For instance, in the [storybook version](https://react.carbondesignsystem.com/iframe.html?args=&id=components-containedlist--with-icons&viewMode=story), I cannot see how any of the icons provided would not be decorative add-ons to list items that might be fruit, plants, berries or fish.

![image](https://github.com/carbon-design-system/carbon/assets/14298245/fd46bd98-dcee-44f7-9eaa-9a21c6aa5bee)



The only scenario I can think of is if there was a list of shopping purchases and the icons were added to designate the food type. But even then, unless there were two identically named list items and the icon provided actual information to distinguish them, it seems likely to be decorative.
Are there examples?

**Comment by andreancardona** [comment]
After discussing this with the team - it is confirmed this is only a descriptive icon and does not provide meaningful information. This not a violation - therefore, closing this issue. Thank you for the help @mbgower!
